### ISSUE TITLE
[x86][R2R][GCStress=3] Assert failure(!"Bad opcode"  CORECLR! LazyMachState::unwindLazyState)

### ISSUE THREAD

**Issue by RussKeldorph** [dotnet/coreclr#5464]
**Failing tests**
JIT._Directed_coverage_oldtests_tlstest_il_d_tlstest_il_d_._Directed_coverage_oldtests_tlstest_il_d_tlstest_il_d_cmd
JIT._Directed_coverage_oldtests_tlstest_il_r_tlstest_il_r_._Directed_coverage_oldtests_tlstest_il_r_tlstest_il_r_cmd

**Environment**
Windows x86 Checked (JIT32)
set COMPlus_GCStress=3
set RunCrossGen=1

**Output**
```
BEGIN EXECUTION
" O:\tfs\pk\src\NDP\clr\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root\crossgen.exe" /Platform_Assemblies_Paths O:\tfs\pk\src\NDP\clr\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root;O:\tfs\pk\src\NDP\clr\bin\tests\Windows_NT.x86.Checked\JIT\Directed\coverage\oldtests\tlstest_il_d\ tlstest_il_d.exe
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Native image O:\tfs\pk\src\NDP\clr\bin\tests\Windows_NT.x86.Checked\JIT\Directed\coverage\oldtests\tlstest_il_d\tlstest.ni.exe generated successfully.
 "O:\tfs\pk\src\NDP\clr\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root\corerun.exe" tlstest_il_d.exe
12

Assert failure(PID 42588 [0x0000a65c], Thread: 23352 [0x5b38]): !"Bad opcode"

CORECLR! LazyMachState::unwindLazyState + 0x3B0 (0x5066e251)
CORECLR! HelperMethodFrame::InsureInit + 0x124 (0x50327f9f)
CORECLR! HelperMethodFrameConfirmState + 0xAF (0x503a3e52)
CORECLR! HelperMethodFrameRestoreState + 0x12 (0x5019642d)
CORECLR! JIT_GetSharedNonGCThreadStaticBase + 0x21D (0x5026453d)
CORECLR! JIT_StaticFieldAddress_Dynamic + 0x24 (0x50279334)
TLSTEST.NI! COM+_Entry_Point + 0xFFFFFFFF (0x73662576)
CORECLR! CallDescrWorkerInternal + 0x34 (0x501963f1)
CORECLR! CallDescrWorker + 0xDB (0x503d9e55)
CORECLR! CallDescrWorkerWithHandler + 0x13C (0x503d9ff6)
    File: o:\tfs\pk\src\ndp\clr\src\vm\i386\gmsx86.cpp Line: 1222
    Image: O:\tfs\pk\src\NDP\clr\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root\CoreRun.exe

Expected: 100
Actual: 1
END EXECUTION - FAILED
FAILED
```

**Comment by RussKeldorph** [comment]
@gkhanna79 @jashook FYI

**Comment by gkhanna79** [comment]
@RussKeldorph Bad opcode in that section means that our "manual" x86 unwinder has encountered a native instruction that it didnt expect. Does this repro with IL?

**Comment by RussKeldorph** [comment]
@gkhanna79 It does not repro without RunCrossGen set.

**Comment by gkhanna79** [comment]
@RussKeldorph Okay, so no repro with IL only runs under GCstress, right? Do you have a remote available?

**Comment by JohnChen0** [comment]
Resolved with #5491 and #5494.
